php-classes/Slate/CBL/StudentCompetency.php

```diff
@@ -326,13 +328,35 @@ public function isLevelComplete()
         $competencyEvidenceRequirements = $this->Competency->getTotalDemonstrationsRequired($this->Level);
         $minimumOffset = $this->Competency->getMinimumAverageOffset();
 
-        return (
-            $completed >= $competencyEvidenceRequirements &&
-            (
-                $logged === 0 ||
-                $average >= ($this->Level + $minimumOffset)
-            )
-        );
+        // Require a minimum total demonstrations for the competency
+        if ($competencyEvidenceRequirements && $completed < $competencyEvidenceRequirements) {
+            return false;
+        }
+
+        // Require minimum average as offset from level
+        if ($minimumOffset !== null && $average < $this->Level + $minimumOffset) {
+            return false;
+        }
+
+        // Require all demonstrations have ratings above minimum
+        if (static::$minimumRatingOffset !== null) {
+            $minimumRating = $this->Level - static::$minimumRatingOffset;
```

Comment:
@jtwiest i renamed this variable to `$minimumRating`, it's best to avoid abbrevations